repo_health/index/static/client/src/main.module.js

```diff
@@ -25,20 +29,54 @@ export const main = angular.module('repo-health', [
     uiRouter,
     ngMaterial,
     components,
-    resources
+    resources,
+    'chart.js'
   ])
   .component('main', mainComponent)
-  .config(($locationProvider, $urlRouterProvider, $mdThemingProvider) => {
+  .config(($locationProvider, $urlRouterProvider, $mdThemingProvider, $stateProvider) => {
     'ngInject';
-    
     $locationProvider.hashPrefix('');
     $urlRouterProvider.otherwise('/search');
+    $stateProvider.state('repo-health', {
+        url: '/repo-health/:owner__login/:name',
+        resolve: {
+            statsUrls: (RepoDetailsService, $stateParams, $state) => {
+                return RepoDetailsService.getStatsUrls($stateParams)
+                    .then(resp => {return resp;}, () => { return {error:true}});
+            }
+        },
+        views: {
+            'repo-details': {
+                controller:  RepoDetailsController,
```

Comment:
Why do you need the controllers. The component should be using the controllers. Also, you can do `$resolve.detailsUrl` in the template
```js
 '<repo-details details-url="$resolve.detailsUrl"></repo-details>'
```